### PR TITLE
Add 'type' option for .as() command; Improve visibility of alias changes in 12.x

### PR DESCRIPTION
**Pull request by BlueWinds**
Adding documentation for the new `type` option for aliases.

This PR also goes back and makes sure the alias changes are listed as 'breaking' in the 12.0.0 changelog. The information was available in the migration guide, but needs more visibility based on the number of users encountering it unexpectedly.

Matching code PR: https://github.com/cypress-io/cypress/pull/25251

### CODE REVIEW

File: content/_changelogs/12.0.0.md [hunk 1]
```diff
@@ -94,6 +94,9 @@ Read more about 12.0 in
   [`cy.visit()`](/api/commands/visit) command generates urls with query
   parameters. Addressed in
   [#20302](https://github.com/cypress-io/cypress/pull/20302).
+- Aliases created with `.as()` now re-run queries leading up to them. To opt in
+  to the old behavior, where aliases are static once defined, use
+  `.as('aliasName', { type: 'value' })`, introduced in Cypress 12.3.0.
```
Comment (emilyrohrbough) [inline content/_changelogs/12.0.0.md:99]:
```suggestion
  `.as('aliasName', { type: 'value' })`, introduced in Cypress [12.3.0](#12-3-0).
```

File: content/guides/references/migration-guide.md [hunk 2]
```diff
@@ -207,6 +207,16 @@ alias to be the first button rather than the unexpanded button.
 cy.findByTestId('popover').findAllByRole('button').first().as('button')
 ```
 
+If you want to alias a static value, such that it is never re-queried, you will
+need Cypress 12.3.0 or later, which introduced the `type` option for `.as()` to
```
Comment (emilyrohrbough) [inline content/guides/references/migration-guide.md:211]: Need to add link to changelog & to `.as()` command.

File: content/api/commands/as.md [hunk 3]
```diff
@@ -44,6 +46,14 @@ The name of the alias to be referenced later within a
 [`cy.get()`](/api/commands/get) or [`cy.wait()`](/api/commands/wait) command
 using an `@` prefix.
 
+**<Icon name="angle-right"></Icon> options** **_(Object)_**
+
+Pass in an options object to change the default behavior of `.as`.
```
Comment (emilyrohrbough) [inline content/api/commands/as.md:51]:
```suggestion
Pass in an options object to change the default behavior of `.as()`.
```

File: content/guides/references/migration-guide.md (hunk 2 above)
Comment (BlueWinds) [reply content/guides/references/migration-guide.md]: Added.

File: content/api/commands/as.md [hunk 4]
```diff
@@ -44,6 +46,14 @@ The name of the alias to be referenced later within a
...
+
+| Option | Default | Description                                                                                                                                                                                                                          |
+| ------ | ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `type` | `query` | Valid values are `query` and `value`. A query alias re-runs all queries leading up to it each time the value is requested, providing a live value. A value alias is frozen as soon as the alias is stored, providing a static value. |
```
[the remark this answers is not in the document]
Comment (BlueWinds) [reply content/api/commands/as.md]: Added a sentence explaining that these don't have any effect referencing spies / stubs / intercepts. Also updated to `query` / `static` to match change in the code PR.

File: docs/api/queries/as.mdx [hunk 5]
```diff
@@ -45,6 +47,14 @@ The name of the alias to be referenced later within a
...
+| ------ | ------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `type` | `query` | The type of alias to store, which impacts how the value is retrieved later in the test. Valid values are `query` and `static`. A `query` alias re-runs all queries leading up to the resulting value each time the alias is requested. A `static` alias is retrived once when the alias is stored, and will never change. `type` has no effect when aliasing intercepts, spies and stubs. |
```
Comment (chrisbreiding) [inline docs/api/queries/as.mdx:56]:
```suggestion
| `type` | `query` | The type of alias to store, which impacts how the value is retrieved later in the test. Valid values are `query` and `static`. A `query` alias re-runs all queries leading up to the resulting value each time the alias is requested. A `static` alias is retrieved once when the alias is stored, and will never change. `type` has no effect when aliasing intercepts, spies, and stubs. |
```

**Review by chrisbreiding — CHANGES_REQUESTED** [review]

**Review by chrisbreiding — APPROVED** [review]

File: docs/guides/references/_changelogs/12.0.0.md [hunk 6]
```diff
@@ -94,6 +94,10 @@ Read more about 12.0 in
   [`cy.visit()`](/api/commands/visit) command generates urls with query
   parameters. Addressed in
   [#20302](https://github.com/cypress-io/cypress/pull/20302).
+- Aliases created with `.as()` now re-run queries leading up to them. To opt in
```
Comment (emilyrohrbough) [inline docs/guides/references/_changelogs/12.0.0.md:97]:
```suggestion
- Aliases created with `.as()` now re-run queries leading up to them, which is equivalent to `.as({ type: 'query' })`. To opt in
```

File: docs/guides/references/migration-guide.mdx [hunk 7]
```diff
@@ -239,6 +239,17 @@ alias to be the first button rather than the unexpanded button.
 cy.findByTestId('popover').findAllByRole('button').first().as('button')
 ```
 
+If you want to alias a static value, such that it is never re-queried, you will
+need Cypress [12.3.0](/guides/references/changelog#12-3-0) or later, which
+introduced the `type` option for [`.as()`](/api/commands/as) to opt into the old
+behavior.
+
+```
+cy.get('.username').invoke('val').as('username', { type: 'value' })
```
Comment (emilyrohrbough) [inline docs/guides/references/migration-guide.mdx:248]:
```suggestion
cy.get('.username').invoke('val').as('username', { type: 'static' })
```

File: docs/api/queries/as.mdx [hunk 8]
```diff
@@ -26,6 +27,7 @@ command with an `@` prefix.
 
 ```javascript
 cy.get('.main-nav').find('li').first().as('firstNav') // Alias element as @firstNav
+cy.get('input.username').invoke('val').as('username', { type: 'static' }) // Alias a static (not re-queried) value
```
Comment (emilyrohrbough) [inline docs/api/queries/as.mdx:30]:
```suggestion
cy.get('input.username').invoke('val').as('username', { type: 'static' }) // Alias that references the value at the time the alias was created
```

**Pull request merged**